client/app/pages/queries/hooks/useDeleteVisualization.js

```diff
@@ -0,0 +1,22 @@
+import { clone, extend, filter, isFunction } from "lodash";
+import { useRef, useCallback } from "react";
+import { Visualization } from "@/services/visualization";
+import notification from "@/services/notification";
+
+export default function useDeleteVisualization(query, onChange) {
+  const onChangeRef = useRef(null);
+  onChangeRef.current = isFunction(onChange) ? onChange : () => {};
```

Comment:
It would be interesting to create a custom hook to do this (should make it clearer on what's it for). But it's still only 2 lines and it has to return the ref itself as it needs to be an object, benefits are nothing, but the the naming.